lib/src/over_react_redux/devtools/middleware.dart

```diff
@@ -0,0 +1,158 @@
+      if (shouldRethrow) rethrow;
+    }
+  }
+
+  dynamic _encodeActionForTransit(dynamic action) {
+    try {
+      return _encodeForTransit({'type': _getActionType(action), 'payload': action}, shouldRethrow: true);
+    } on Error {
+      return _encodeForTransit({'type': _getActionType(action)});
+    }
+  }
+
+  String _getActionType(dynamic action) {
+    if (action.toString().contains('Instance of')) {
+      return action.runtimeType.toString();
+    }
+    return action.toString();
+  }
+
+  _relay(String type, [Object state, dynamic action, String nextActionId]) {
+    final message = JsBackedMap();
+    message['type'] = type;
+
+    if (state != null) {
+      try {
+        message['payload'] = _encodeForTransit(state, shouldRethrow: true);
+      } catch (_) {
+        message['payload'] = 'Could not encode state. Ensure state is json encodable';
+      }
+    }
+
+    if (type == 'ACTION') {
+      message['action'] = _encodeActionForTransit(action);
+      message['nextActionId'] = nextActionId;
+    } else if (action != null) {
+      message['action'] = action as String;
+    }
+
+    if (message.containsKey('action')) {
+      devToolsExt.send(message['action'], message['payload']);
+    } else {
+      devToolsExt.send(message.jsObject);
+    }
+  }
+
+  void handleEventFromRemote(JsMap jsData) {
+    JsBackedMap data = JsBackedMap.fromJs(jsData);
+    switch (data['type'] as String) {
+      case 'DISPATCH':
+        _handleDispatch(JsBackedMap.fromJs(data['payload']));
+        break;
+      case 'ACTION':
+        _handleRemoteAction(data['action'] as String);
+        break;
+      default:
+        log.warning('Unknown type:' + data['type'].toString());
+    }
+  }
+
+  void _handleDispatch(dynamic action) {
+    if (this._store == null) {
+      log.warning('No store reference set, cannot dispatch remote action');
+      return;
+    }
+    switch (action['type'] as String) {
+      case 'JUMP_TO_ACTION':
+      case 'JUMP_TO_STATE':
+        this._store.dispatch(DevToolsAction.jumpToState(action['actionId'] as int));
+        break;
+      default:
+        log.warning("Unknown command: ${action['type']}. Ignoring");
+    }
+  }
+
+  void _handleRemoteAction(String action) {
+    if (this._store == null) {
+      log.warning('No store reference set, cannot dispatch remote action');
+      return;
+    }
+    this._store.dispatch(DevToolsAction.perform(jsonDecode(action)));
+  }
+
+  /// Middleware function called by redux, dispatches actions to devtools
+  @override
+  call(Store store, dynamic action, NextDispatcher next) {
+    next(action);
+    if (devToolsExt == null) return;
+    this.store ??= store;
+    if (!(action is DevToolsAction)) {
+      this._relay('ACTION', store.state, action);
+    }
+  }
+}
+
+final overReactReduxDevToolsMiddleware = OverReactReduxDevToolsMiddleware();
```

Comment:
To reduce the surface area of what API we commit to, could we just export this? (And not let the type leak if necessary?)

```suggestion
final MiddlewareClass overReactReduxDevToolsMiddleware = OverReactReduxDevToolsMiddleware();
```